### PR TITLE
Add reader and writer custom HTTP headers flags

### LINKED ISSUE #1020: Add headers support in Prom-migrator
In prom-migrator add support for headers to writer URL, helps in configuring tenant and auth-related details through headers. 

```
./prom-migrator_0.0.3_Linux_x86_64 -start=1641486391
-reader-url=http://x.x.x.x:portnumber/prom/api/v1/read
-writer-url=http://y.y.y.y:portnumber/write -progress-enabled=false
**--headers="TENANT=<>"**
```

### PR DESCRIPTION
**Pull request by alejandrodnm**
## Description

The 2 new flags `reader-http-header` and `writer-http-header` allow
users to specify custom HTTP headers for the reader and writter
requests respectively.

The format for the flag values is `key:value`. These flags can be set
multiple times to specify multiple headers, or to set multiple values
for the same header:

- Multiple headers:

  `-writer-http-header="X-Org-Id:42" -writer-http-header="Foo:bar"`

- Multiple values for the same header:

  `-reader-http-header="X-Org-Id:42" -reader-http-header="X-Org-Id:24"`

closes #1020 

## Merge requirements

Please take into account the following non-code changes that you may need to make with your PR:

- [ ] CHANGELOG entry for user-facing changes
- [x] Updated the relevant documentation

### CODE REVIEW

**Comment by alejandrodnm** [comment]
Do we add CHANGELOG entries for changes in prom-migrator?

File: migration-tool/cmd/prom-migrator/main.go [hunk 1]
```diff
@@ -181,6 +182,34 @@ loop:
...
+	var s string
+	for k, vs := range f.headers {
+		s += k + "=" + strings.Join(vs, ",")
+	}
+	return s
+}
+
+func (f headersFlags) Set(value string) error {
+	k, v, _ := strings.Cut(value, ":")
+	if k == "" {
```
Comment (niksajakovljevic) [inline migration-tool/cmd/prom-migrator/main.go:199]: You probably want to check `v` here, not `k` (b/c k will contain the actual value)

Comment (niksajakovljevic) [reply migration-tool/cmd/prom-migrator/main.go:199]: Or just check 3rd bool

**Review by niksajakovljevic — APPROVED** [review]

File: migration-tool/cmd/prom-migrator/main.go [hunk 2]
```diff
@@ -181,6 +182,34 @@ loop:
 	log.Info("msg", "exiting!")
 }
 
+type headersFlags struct {
```
Comment (niksajakovljevic) [inline migration-tool/cmd/prom-migrator/main.go:185]: Nit: Consider moving this outside of main since it deals with very specific things

**Review by Harkishen-Singh — APPROVED** [review]
Some nits.

**Pull request merged**